### PR TITLE
Deserialize WindowsGL & Linux -> DesktopGL

### PR DESCRIPTION
**Pull request by nkast**
...and remove them from TargetPlatform enumeration.

Fix for #4154

### CODE REVIEW

File: MonoGame.Framework.Content.Pipeline/Builder/PathHelper.cs [hunk 1]
```diff
@@ -65,7 +65,7 @@ public static string NormalizeOS(string path, TargetPlatform targetPlatform)
             switch (targetPlatform)
             {
                 case TargetPlatform.Windows:
-                case TargetPlatform.WindowsGL:
+                case TargetPlatform.DesktopGL:
```
Comment (cra0zy) [inline MonoGame.Framework.Content.Pipeline/Builder/PathHelper.cs]: Remove this line completely, it shouldn't have been there in the first place.

Comment (nkast) [inline MonoGame.Framework.Content.Pipeline/Builder/PathHelper.cs]: I's rather not alter the current behavior with a PR that's unrelated to PathHelper.

**Comment by tomspilman** [comment]
So does this also fix the platform dropdown in the Pipeline tool?  What appears in that list now?

**Comment by cra0zy** [comment]
> So does this also fix the target platform dropdown in the Pipeline tool? What appears in that list now?

So, this does the following:
 - Pipeline Tool will only display DesktopGL
 - Pipeline Tool will not display WindowsGL and Linux
 - if Pipeline Tool loads something with WindowsGL or Linux platforms it will move it to DesktopGL platform
 - if MGCB comes across WindowsGL or Linux it will compile it as if it was DesktopGL

**Comment by tomspilman** [comment]
> Pipeline Tool will not display WindowsGL and Linux

So it still shows MacOS?

**Comment by cra0zy** [comment]
> So it still shows MacOS?

Yes... MacOS as a platform is same as WindowsDX... both have nothing to do with DesktopGL...

**Comment by tomspilman** [comment]
> Yes... MacOS as a platform is same as WindowsDX... both have nothing to do with DesktopGL...

Why is that?  Shouldn't MacOS be a part of DesktopGL now?

**Comment by hach-que** [comment]
DesktopGL supports Mac, but it doesn't support it actually being a Mac OS application. i.e. DesktopGL is just a normal .NET executable on Mac, it's not a Mac .app package (nor can you publish them to the Mac app store).

**Comment by mrhelmut** [comment]
You can package DesktopGL projects to .app with MonoKickstart, that works just fine if one plans to publish through self hosting / Steam, but I don't know if such .app are Mac AppStore compliant (that's the whole purpose of Xamarin Mac and the new Mac API).

Both are valid paths IMO.

**Comment by cra0zy** [comment]
Well, I don't think there is anything left for me to say, you guys said it all...

**Comment by tomspilman** [comment]
> Both are valid and different paths IMO.

Ahh... ok... so this is sort of like the difference in a Windows desktop app and a Windows Store app.  Technically the same platform, but just different packaging methods for those platforms.  That whole concept makes this `TargetPlatform` enum really weird to understand.

**Comment by cra0zy** [comment]
> Ahh... ok... so this is sort of like the difference in a Windows desktop app and a Windows Store app. Technically the same platform, but just different packaging methods for those platforms. That whole concept makes this TargetPlatform enum really weird to understand.

We should probably rename some TargetPlatform enums:
 - Windows > WindowsDX
 - MacOS > XamMac

**Comment by tomspilman** [comment]
> We should probably rename some TargetPlatform enums:

I'm hesitant to do that until we have a bigger discussion about it.  Mostly it just seems like what is presented to the user via the Pipeline tool is really confusing as it is now.

**Comment by tomspilman** [comment]
Thanks @nkast and @cra0zy !

**Pull request merged**